Library/Homebrew/cmd/update.sh

```diff
@@ -566,23 +568,38 @@ EOS
         echo "Fetching $DIR..."
       fi
 
+      local tmp_failure_file="$HOMEBREW_REPOSITORY/.git/TMP_FETCH_FAILURES"
+      rm -f "$tmp_failure_file"
+
       if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
       then
         git fetch --tags --force "${QUIET_ARGS[@]}" origin \
           "refs/heads/$UPSTREAM_BRANCH_DIR:refs/remotes/origin/$UPSTREAM_BRANCH_DIR" 2>/dev/null
       else
+        # Capture stderr to tmp_failure_file
         if ! git fetch --tags --force "${QUIET_ARGS[@]}" origin \
-          "refs/heads/$UPSTREAM_BRANCH_DIR:refs/remotes/origin/$UPSTREAM_BRANCH_DIR"
+          "refs/heads/$UPSTREAM_BRANCH_DIR:refs/remotes/origin/$UPSTREAM_BRANCH_DIR" 2>>"$tmp_failure_file"
         then
+          # Reprint fetch errors to stderr
+          [[ -f "$tmp_failure_file" ]] && cat "$tmp_failure_file" 1>&2
+
           if [[ "$UPSTREAM_SHA_HTTP_CODE" = "404" ]]
           then
             TAP="${DIR#$HOMEBREW_LIBRARY/Taps/}"
             echo "$TAP does not exist! Run \`brew untap $TAP\` to remove it." >>"$update_failed_file"
           else
             echo "Fetching $DIR failed!" >>"$update_failed_file"
+
+            if [[ -f "$tmp_failure_file" ]] &&
+               [[ "$(<"$tmp_failure_file")" = "fatal: couldn't find remote ref refs/heads/$UPSTREAM_BRANCH_DIR" ]]
+            then
+              echo "$DIR" >>"$failed_fetch_dirs_file"
```

Comment:
```suggestion
              echo "$DIR" >>"$missing_remote_ref_dirs_file"
```

perhaps if it's now only being used in this case. Out of interest: what's the error if the repository no longer exists at all? Should be fairly easy to reproduce by editing `.git/config` to have an invalid repository URL.

Reply:
Here's the output if the repo doesn't exist. Currently, this is handled separately so the message will not show (there's a different one that does). I don't think there's anything `brew tap --repair` can do in this situation.

```console
$ brew update
remote: Repository not found.
fatal: repository 'https://github.com/rylan12/homebrew-branch-name-test1/' not found
Error: rylan12/homebrew-branch-name-test does not exist! Run `brew untap rylan12/homebrew-branch-name-test` to remove it.
```

Reply:
Perfect, thanks!